tasty-query/shared/src/main/scala/tastyquery/Subtyping.scala

```diff
@@ -0,0 +1,306 @@
+package tastyquery
+
+import tastyquery.Contexts.*
+import tastyquery.Exceptions.*
+import tastyquery.Flags.*
+import tastyquery.Symbols.*
+import tastyquery.Types.*
+import tastyquery.TypeMaps.*
+
+private[tastyquery] object Subtyping:
+  def isSubtype(tp1: Type, tp2: Type)(using Context): Boolean =
+    level1(tp1, tp2)
+
+  private def level1(tp1: Type, tp2: Type)(using Context): Boolean = tp2 match
+    case tp2: NamedType =>
+      // Try follow type alias
+      val sym2 = tp2.symbol
+      sym2 match
+        case sym2: TypeMemberSymbol if sym2.isTypeAlias =>
+          if isSubtype(tp1, sym2.aliasedTypeAsSeenFrom(tp2.prefix)) then return true
+        case _ =>
+          ()
+      end match
+
+      tp1 match
+        case tp1: NamedType =>
+          val sym1 = tp1.symbol
+          sym1 match
+            case sym1: TypeMemberSymbol if sym1.isTypeAlias =>
+              if isSubtype(sym1.aliasedTypeAsSeenFrom(tp1.prefix), tp2) then return true
+            case _ =>
+              ()
+          end match
+          level1NamedNamed(tp1, tp2)
+
+        case _ =>
+          level2(tp1, tp2)
+
+    case tp2: WildcardTypeBounds =>
+      isSubtype(tp1, tp2.bounds.high)
+
+    case tp2: ThisType =>
+      val cls2 = tp2.cls
+      tp1 match
+        case tp1: ThisType =>
+          tp1.cls == cls2
+        case tp1: NamedType if cls2.is(Module) && cls2 == tp1.symbol =>
+          (cls2.isStatic || isSubprefix(tp1.prefix, cls2.typeRef.prefix))
+            || level2(tp1, tp2)
+        case _ =>
+          level2(tp1, tp2)
+
+    case tp2: AndType =>
+      isSubtype(tp1, tp2.first) && isSubtype(tp1, tp2.second)
+
+    case tp2: ConstantType =>
+      tp1 match
+        case tp1: ConstantType =>
+          tp1.value == tp2.value
+        case _ =>
+          level2(tp1, tp2)
+
+    case _ =>
```

Comment:
I don't see an equivalent of https://github.com/lampepfl/dotty/blob/ef962cc2ee8d7cb2e8aebb5391406c74465e29c1/compiler/src/dotty/tools/dotc/core/TypeComparer.scala#L344-L345